include/aspect/material_model/interface.h

```diff
@@ -840,7 +840,7 @@ namespace aspect
          * 0, then the viscosity does not need to be computed.
          */
         virtual void evaluate(const MaterialModel::MaterialModelInputs<dim> &in,
-                              MaterialModel::MaterialModelOutputs<dim> &out) const = 0;
+                              MaterialModel::MaterialModelOutputs<dim> &out) const;
```

Comment:
Essentially you already solved it. You just also need to remove the default implementation of the `evaluate` function in the `interface.cc` file (remove lines 103-109). This forces every derived class (i.e. each material model) to implement this function, otherwise it will not compile. We do this, because nothing good can happen if a material model does not implement this function.